File: cpp/Sorting Algorithms/Bubble Sort.cpp
#include <iostream>
using namespace std;

void swap(int &a, int &b)
{
	int temp = a;
	a = b;
	b = temp;
}

void bubbleSort(int *arr, int n)
{
	for (int i = 0; i < n; i++)
		for (int j = 0; j < n-i-1; j++)
			if (arr[j] > arr[j+1])
				swap(arr[j], arr[j+1]);
}

int main()
{
	int n;
	cout << "Enter the number of Elements: ";
	cin >> n;
	int arr[n];
	cout << "\nEnter the Array: ";
	for (int i = 0; i < n; i++)
		cin >> arr[i];

	bubbleSort(arr, n);

	cout << "\nThe Array after sorting is: ";
	for (int i = 0; i < n; i++)
		cout << arr[i] << " ";
}

File: JavaScript/BubbleSort/bubbleSort.js
var bubbleSort = (inputArray) => {
    let swap;
    var swapped;
    do{
        swapped = false;
        for(let i = 0; i<inputArray.length; i++){
            if(inputArray[i] > inputArray[i+1]){
                var temp = inputArray[i];
                inputArray[i] = inputArray[i+1];
                inputArray[i+1] = temp;
                swapped = true;
                }
            }
    }
    while(swapped);
    return inputArray;
};

a = [3,5,71,2,65,44];
console.log(bubbleSort(a));